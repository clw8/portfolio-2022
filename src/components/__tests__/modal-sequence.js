import {act } from '@testing-library/react';
import React from "react"

import ModalSequence from "../modal-sequence"
import Modal from '../modal';
import getInstance from '../../utils/test-utils/get-instance';
import "@testing-library/jest-dom"



describe("Modal Sequence", () => {
  let renderInstance;

  beforeEach(() => {
    renderInstance = getInstance(ModalSequence, {
      data: ["test data here", "second modal", "third modal", 3, 4], 
      renderModal: ({show, index, datum, onNext, onPrevious, exitModalSequence}) => {
        return (
          <Modal 
            show={show}
            key={index}
          >
            <button onClick={onPrevious}>Previous</button>
            <p>{datum}</p>
            <button onClick={onNext}>Next</button>
            <button onClick={exitModalSequence}>Close</button>
          </Modal>
      )}})
    
  })

  it("renders correctly", () => {
    expect(renderInstance.container).toBeInTheDocument()
    const firstRender = renderInstance.asFragment()
    expect(firstRender).toMatchSnapshot()
  })

  it("renders the modals correctly ", () => {
    expect(renderInstance.container).toHaveTextContent("test data here")
    expect(renderInstance.container).toHaveTextContent("second modal")
    expect(renderInstance.container).toHaveTextContent("Previous")
    expect(renderInstance.container).toHaveTextContent("Next")
  })

  it("can go to the next and previous modals", () => {
    act(() => {
      renderInstance.instanceRef.current.goToIndex(1)
    })
    const shownModal = renderInstance.getByRole('dialog')
    expect(shownModal).toBeInTheDocument()
    expect(shownModal).toHaveTextContent("second modal")
    act(() => {
      renderInstance.instanceRef.current.goToNext()
    })
    const nextModal = renderInstance.getByRole('dialog')
    expect(nextModal).toHaveTextContent("third modal")
    
    act(() => {
      renderInstance.instanceRef.current.goToPrevious()
    })

    act(() => {
      renderInstance.instanceRef.current.goToPrevious()
    })

    const firstModal = renderInstance.getByRole('dialog')
    expect(firstModal).toHaveTextContent("test data here")

  })
})