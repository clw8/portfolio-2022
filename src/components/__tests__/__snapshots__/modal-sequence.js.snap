// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Sequence renders correctly 1`] = `
<DocumentFragment>
  <div
    animation="out-right-end"
    aria-hidden="true"
    class="modal"
    role="dialog"
  >
    <div
      class="modal__inner"
    >
      <button>
        Previous
      </button>
      <p>
        test data here
      </p>
      <button>
        Next
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
  <div
    animation="out-right-end"
    aria-hidden="true"
    class="modal"
    role="dialog"
  >
    <div
      class="modal__inner"
    >
      <button>
        Previous
      </button>
      <p>
        second modal
      </p>
      <button>
        Next
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
  <div
    animation="out-right-end"
    aria-hidden="true"
    class="modal"
    role="dialog"
  >
    <div
      class="modal__inner"
    >
      <button>
        Previous
      </button>
      <p>
        third modal
      </p>
      <button>
        Next
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
  <div
    animation="out-right-end"
    aria-hidden="true"
    class="modal"
    role="dialog"
  >
    <div
      class="modal__inner"
    >
      <button>
        Previous
      </button>
      <p>
        3
      </p>
      <button>
        Next
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
  <div
    animation="out-right-end"
    aria-hidden="true"
    class="modal"
    role="dialog"
  >
    <div
      class="modal__inner"
    >
      <button>
        Previous
      </button>
      <p>
        4
      </p>
      <button>
        Next
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</DocumentFragment>
`;
